src/services/factories/getTokenFromExchange.ts

```diff
@@ -49,23 +49,38 @@ interface GetByAddressParams {
   tokenAddress: string
 }
 
+export const enum TokenFromExchange {
```

Comment:
I can't guess what this enum models reading it's name.
Looks like it's used for documenting a "reason" of the value in token? 

I mean, shouldn't it return the data, unless there's a problem like "Unregistered" and in that case you reject providing that reason? why do we want to fulfill the promise in that case?

Reply:
I added reason to be able to distinguish why token couln't be fetched.
As when fetching to early, which happens when opening modal right away with token from URL, we may be between providers (switching to lastUsedProvider). And it fails to fetch fro ExchangeApi
Then we retry. But if it failed for any other reason then no use retrying.

Reply:
Can of course throw an error, but then those can be different errors and we'll still need to differentiate the reason

Reply:
if it's easier for you go ahead, although in principle you can throw an error with a type field, that would allow to catch and retry depending on the type